packages/e2e/test/stake-pool/StakePoolProvider.test.ts

```diff
@@ -0,0 +1,220 @@
+/* eslint-disable max-len */
+/* eslint-disable sonarjs/cognitive-complexity */
+import * as envalid from 'envalid';
+import { Cardano, StakePoolProvider } from '@cardano-sdk/core';
+import { logger } from '@cardano-sdk/util-dev';
+import { stakePoolHttpProvider } from '@cardano-sdk/cardano-services-client';
+
+describe('stake-pool queries', () => {
+  const pagination = { limit: 20, startAt: 0 };
+
+  let provider: StakePoolProvider;
+  let pools: Cardano.StakePool[] = [];
+
+  beforeAll(() => {
+    const env = envalid.cleanEnv(process.env, { STAKE_POOL_PROVIDER_URL: envalid.url() });
+    const config = { baseUrl: env.STAKE_POOL_PROVIDER_URL, logger };
+
+    provider = stakePoolHttpProvider(config);
+  });
+
+  it('without filters, some pools is found', async () => {
+    const { pageResults, totalResultCount } = await provider.queryStakePools({ pagination });
+    expect(totalResultCount).toBeGreaterThan(0);
+    pools = pageResults;
+  });
+
+  describe('filters and options', () => {
+    beforeAll(() => {
+      if (pools.length < 2)
+        throw new Error(
+          `'some pools is found' test didn't run or found only ${pools.length} pools, but at least 2 pools are required to run this test`
+        );
+    });
+
+    describe('identifier filter', () => {
+      it('using id as value', async () => {
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ id: pools[0].id }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using name as value', async () => {
+        const pool = pools.find(({ metadata }) => Boolean(metadata?.name));
+
+        if (!pool)
+          throw new Error("'some pools is found' test found only pools without name: this test can't be performed");
+
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ name: pool.metadata!.name }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using ticker as value', async () => {
+        const pool = pools.find(({ metadata }) => Boolean(metadata?.ticker));
+
+        if (!pool)
+          throw new Error("'some pools is found' test found only pools without ticker: this test can't be performed");
+
+        const { totalResultCount } = await provider.queryStakePools({
+          filters: { identifier: { values: [{ ticker: pool.metadata!.ticker }] } },
+          pagination
+        });
+        expect(totalResultCount).toBe(1);
+      });
+
+      it('using id OR ticker as value (same pool)', async () => {
```

Comment:
Would be nice to find and query 3 pools here:
1. has ticker and name
2. has ticker, no name
3. no ticker, has name

Reply:
Sorry @mkazlauskas I have not clear the test.

Could it be you are speaking about a test which performs a query with three ids in OR, each one of them referring to:

1. a pool with ticker and name
2. a pool with ticker but no name
3. a pool with name, but no ticker

?

Reply:
Yes, that would work